src/openrct2-ui/ride/Construction.h

```diff
@@ -10,5 +10,153 @@
+    TrackElemType::FlyerHalfLoopUninvertedUp,
+    TrackElemType::FlyerHalfLoopInvertedDown,
+    TrackElemType::LeftFlyerCorkscrewUp,
+    TrackElemType::RightFlyerCorkscrewUp,
+    TrackElemType::LeftFlyerCorkscrewDown,
+    TrackElemType::RightFlyerCorkscrewDown,
+    TrackElemType::HeartLineTransferUp,
+    TrackElemType::HeartLineTransferDown,
+    TrackElemType::LeftHeartLineRoll,
+    TrackElemType::RightHeartLineRoll,
+    TrackElemType::MinigolfHoleA,
+    TrackElemType::MinigolfHoleB,
+    TrackElemType::MinigolfHoleC,
+    TrackElemType::MinigolfHoleD,
+    TrackElemType::MinigolfHoleE,
+    TrackElemType::MultiDimInvertedFlatToDown90QuarterLoop,
+    TrackElemType::Up90ToInvertedFlatQuarterLoop,
+    TrackElemType::InvertedFlatToDown90QuarterLoop,
+    TrackElemType::LeftCurvedLiftHill,
+    TrackElemType::RightCurvedLiftHill,
+    TrackElemType::LeftReverser,
+    TrackElemType::RightReverser,
+    TrackElemType::AirThrustTopCap,
+    TrackElemType::AirThrustVerticalDown,
+    TrackElemType::AirThrustVerticalDownToLevel,
+    TrackElemType::BlockBrakes,
+    TrackElemType::MultiDimUp90ToInvertedFlatQuarterLoop,
+    TrackElemType::MultiDimFlatToDown90QuarterLoop,
+    TrackElemType::MultiDimInvertedUp90ToFlatQuarterLoop,
+    TrackElemType::RotationControlToggle,
+    TrackElemType::LeftLargeCorkscrewUp,
+    TrackElemType::RightLargeCorkscrewUp,
+    TrackElemType::LeftLargeCorkscrewDown,
+    TrackElemType::RightLargeCorkscrewDown,
+    TrackElemType::LeftMediumHalfLoopUp,
+    TrackElemType::RightMediumHalfLoopUp,
+    TrackElemType::LeftMediumHalfLoopDown,
+    TrackElemType::RightMediumHalfLoopDown,
+    TrackElemType::LeftZeroGRollUp,
+    TrackElemType::RightZeroGRollUp,
+    TrackElemType::LeftZeroGRollDown,
+    TrackElemType::RightZeroGRollDown,
+    TrackElemType::LeftLargeZeroGRollUp,
+    TrackElemType::RightLargeZeroGRollUp,
+    TrackElemType::LeftLargeZeroGRollDown,
+    TrackElemType::RightLargeZeroGRollDown,
+    TrackElemType::LeftFlyerLargeHalfLoopUninvertedUp,
+    TrackElemType::RightFlyerLargeHalfLoopUninvertedUp,
+    TrackElemType::LeftFlyerLargeHalfLoopInvertedDown,
+    TrackElemType::RightFlyerLargeHalfLoopInvertedDown,
+    TrackElemType::LeftFlyerLargeHalfLoopInvertedUp,
+    TrackElemType::RightFlyerLargeHalfLoopInvertedUp,
+    TrackElemType::LeftFlyerLargeHalfLoopUninvertedDown,
+    TrackElemType::RightFlyerLargeHalfLoopUninvertedDown,
+    TrackElemType::FlyerHalfLoopInvertedUp,
+    TrackElemType::FlyerHalfLoopUninvertedDown,
+};
+constexpr const size_t DropdownLength = DropdownOrder.size();
+
+// Reminder to add your track element to the list, if applicable
+static_assert(TrackElemType::Count == 337);
+
+constexpr bool TrackPieceDirectionIsDiagonal(const uint8_t direction)
+{
+    return direction >= NumOrthogonalDirections;
+}
+
+struct SpecialElement
+{
+    track_type_t TrackType;
+    bool Disabled;
+};
+
+struct SpecialElementsList
+{
+    int32_t NumElements{};
```

Comment:
If you need a NumElements i'd prefer you switched Elements to being a std::vector and you grow to NumElements. That makes things a bit more sensible.